docs/src/main/sphinx/connector/faker.md

```diff
@@ -0,0 +1,225 @@
+faker.null-probability=0.1
+faker.default-limit=1000
+```
+
+Create a table with the same columns as in the table to be populated with
+random data. Exclude all properties, because the Faker connector doesn't
+support same properties as other connectors.
+
+```sql
+CREATE TABLE generator.default.customer (LIKE production.public.customer EXCLUDING PROPERTIES);
+```
+
+Insert random data into the original table, by selecting it from the Faker
+connector. Data generated for columns of non-character types cover the whole
+range of that data type, so to make it more practical, adding constraints might
+be necessary.
+
+```sql
+INSERT INTO production.public.customers
+SELECT *
+FROM generator.default.customers
+WHERE
+  born_at BETWEEN CURRENT_DATE - INTERVAL '150' YEAR AND CURRENT_DATE
+  AND age_years BETWEEN 0 AND 150
+LIMIT 100;
+```
+
+To generate even more realistic data, choose specific generators by setting the
+`generator` property on columns. Start with getting the complete definition of a table.
+
+```sql
+SHOW CREATE TABLE production.public.customers;
+```
+
+Modify the output of the previous query and add some column properties.
+
+```sql
+CREATE TABLE generator.default.customer (
+  id UUID NOT NULL,
+  name VARCHAR NOT NULL WITH (generator = '#{Name.first_name} #{Name.last_name}'),
+  address VARCHAR NOT NULL WITH (generator = '#{Address.fullAddress}'),
+  born_at DATE,
+  age_years INTEGER
+);
+```
+
+## Configuration
+
+The following table details all general configuration properties:
+
+:::{list-table} Faker configuration properties
+:widths: 35, 55, 10
+:header-rows: 1
+
+* - Property name
+  - Description
+  - Default
+* - `faker.null-probability`
+  - Default null probability for any column in any table that allows them.
+  - `0.5`
+* - `faker.default-limit`
+  - Default number of rows for each table, when the LIMIT clause is not specified in the query.
+  - `1000`
+:::
+
+A particular data generator is selected based on the column type.
+
+### Character types
+
+Faker supports the following character types:
+
+- `CHAR`
+- `VARCHAR`
+- `VARBINARY`
+
+Columns of those types use a generator producing the [Lorem
+ipsum](https://en.wikipedia.org/wiki/Lorem_ipsum) placeholder text. Unbounded
+columns return a random sentence with 3 to 40 words.
+
+To have more control over the format of the generated data, use the `generator`
+column property. Some examples of valid generator expressions:
+
+- `#{regexify '(a|b){2,3}'}`
+- `#{regexify '\\.\\*\\?\\+'}`
+- `#{bothify '????','false'}`
+- `#{Name.first_name} #{Name.first_name} #{Name.last_name}`
+- `#{number.number_between '1','10'}`
+
+See the Datafaker's documentation for more information about
+[the expression](https://www.datafaker.net/documentation/expressions/) syntax
+and [available providers](https://www.datafaker.net/documentation/providers/).
+
+To test a generator expression, without having to recreate the table, use the
+`random_string(expression)` function:
```

Comment:
What function is that? We dont have docs for that .. at least not in https://trino.io/docs/current/functions/list.html#r

Reply:
It's provided by this connector.

Reply:
As a global function? Do we need to add it to the functions docs then? if not .. dont you need to specify catalog or something before being able to use it?

Reply:
Yes, it's a global function:

```
trino> show functions like 'random_expression';
     Function      | Return Type | Argument Types | Function Type | Deterministic |                      Description                       
-------------------+-------------+----------------+---------------+---------------+--------------------------------------------------------
 random_expression | varchar     | varchar        | scalar        | true          | Generate a random string based on the Faker expression 
(1 row)
```

Reply:
Added to the string functions page and both list of functions.

Reply:
Whoops, I just noticed it should NOT be deterministic. I'll fix this in a separate PR.

Reply:
https://github.com/trinodb/trino/pull/23977

Reply:
cool .. question .. why is the function in the faker codebase if it is a global function? Does it require a catalog with the connector so it works? Its kinda odd how this is set up .. maybe we should discuss how this works and is set up.

Reply:
We already do this, another example is the `trino-geospatial` plugin that only provides functions.

Reply:
Ok .. and the plugin is loaded automatically anyway .. so the only way this will fail is when that plugin is not even bundled .. we need to take this into account for #22597